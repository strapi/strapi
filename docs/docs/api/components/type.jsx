import React from 'react';

export default function Type({ children }) {
  return (
    <span
      style={{
        color: '#017501',
      }}
    >
      &lt;{children}&gt;
    </span>
  );
}
