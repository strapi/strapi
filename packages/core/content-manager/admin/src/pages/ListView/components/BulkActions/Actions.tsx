import * as React from 'react';

import {
  useStrapiApp,
  DescriptionComponentRenderer,
  useTable,
  useQueryParams,
} from '@strapi/admin/strapi-admin';
import { Box, ButtonProps, Flex, Typography } from '@strapi/design-system';
import { WarningCircle } from '@strapi/icons';
import { useIntl } from 'react-intl';

import { useDocumentRBAC } from '../../../../features/DocumentRBAC';
import { useDoc } from '../../../../hooks/useDocument';
import { useDocumentActions } from '../../../../hooks/useDocumentActions';
import { buildValidParams } from '../../../../utils/api';
import { getTranslation } from '../../../../utils/translations';
import {
  DialogOptions,
  DocumentActionButton,
  ModalOptions,
  NotificationOptions,
} from '../../../EditView/components/DocumentActions';

import { PublishAction } from './PublishAction';

import type { BulkActionComponent, ContentManagerPlugin } from '../../../../content-manager';

interface BulkActionDescription {
  dialog?: DialogOptions | NotificationOptions | ModalOptions;
  disabled?: boolean;
  icon?: React.ReactNode;
  label: string;
  onClick?: (event: React.SyntheticEvent) => void;
  /**
   * @default 'default'
   */
  type?: 'icon' | 'default';
  /**
   * @default 'secondary'
   */
  variant?: ButtonProps['variant'];
}

/* -------------------------------------------------------------------------------------------------
 * BulkActionsRenderer
 * -----------------------------------------------------------------------------------------------*/

const BulkActionsRenderer = () => {
  const plugins = useStrapiApp('BulkActionsRenderer', (state) => state.plugins);

  const { model, collectionType } = useDoc();
  const { selectedRows } = useTable('BulkActionsRenderer', (state) => state);

  return (
    <Flex gap={2}>
      <DescriptionComponentRenderer
        props={{
          model,
          collectionType,
          documents: selectedRows,
        }}
        descriptions={(
          plugins['content-manager'].apis as ContentManagerPlugin['config']['apis']
        ).getBulkActions()}
      >
        {(actions) => actions.map((action) => <DocumentActionButton key={action.id} {...action} />)}
      </DescriptionComponentRenderer>
    </Flex>
  );
};

/* -------------------------------------------------------------------------------------------------
 * DefaultBulkActions
 * -----------------------------------------------------------------------------------------------*/

const DeleteAction: BulkActionComponent = ({ documents, model }) => {
  const { formatMessage } = useIntl();
  const { schema: contentType } = useDoc();
  const selectRow = useTable('DeleteAction', (state) => state.selectRow);
  const hasI18nEnabled = Boolean(contentType?.pluginOptions?.i18n);
  const [{ query }] = useQueryParams<{ plugins?: { i18n?: { locale?: string } } }>();
  const params = React.useMemo(() => buildValidParams(query), [query]);
  const hasDeletePermission = useDocumentRBAC('deleteAction', (state) => state.canDelete);
  const { deleteMany: bulkDeleteAction, isLoading } = useDocumentActions();
  const documentIds = documents.map(({ documentId }) => documentId);

  const handleConfirmBulkDelete = async () => {
    const res = await bulkDeleteAction({
      documentIds,
      model,
      params,
    });
    if (!('error' in res)) {
      selectRow([]);
    }
  };

  if (!hasDeletePermission) return null;

  return {
    variant: 'danger-light',
    label: formatMessage({ id: 'global.delete', defaultMessage: 'Delete' }),
    dialog: {
      type: 'dialog',
      title: formatMessage({
        id: 'app.components.ConfirmDialog.title',
        defaultMessage: 'Confirmation',
      }),
      loading: isLoading,
      content: (
        <Flex direction="column" alignItems="stretch" gap={2}>
          <Flex justifyContent="center">
            <WarningCircle width="24px" height="24px" fill="danger600" />
          </Flex>
          <Typography id="confirm-description" textAlign="center">
            {formatMessage({
              id: 'popUpWarning.bodyMessage.contentType.delete.all',
              defaultMessage: 'Are you sure you want to delete these entries?',
            })}
          </Typography>
          {hasI18nEnabled && (
            <Box textAlign="center" padding={3}>
              <Typography textColor="danger500">
                {formatMessage(
                  {
                    id: getTranslation('Settings.list.actions.deleteAdditionalInfos'),
                    defaultMessage:
                      'This will delete the active locale versions <em>(from Internationalization)</em>',
                  },
                  {
                    em: Emphasis,
                  }
                )}
              </Typography>
            </Box>
          )}
        </Flex>
      ),
      onConfirm: handleConfirmBulkDelete,
    },
  };
};

DeleteAction.type = 'delete';

const UnpublishAction: BulkActionComponent = ({ documents, model }) => {
  const { formatMessage } = useIntl();
  const { schema } = useDoc();
  const selectRow = useTable('UnpublishAction', (state) => state.selectRow);
  const hasPublishPermission = useDocumentRBAC('unpublishAction', (state) => state.canPublish);
  const hasI18nEnabled = Boolean(schema?.pluginOptions?.i18n);
  const hasDraftAndPublishEnabled = Boolean(schema?.options?.draftAndPublish);
  const { unpublishMany: bulkUnpublishAction, isLoading } = useDocumentActions();
  const documentIds = documents.map(({ documentId }) => documentId);
  const [{ query }] = useQueryParams();
  const params = React.useMemo(() => buildValidParams(query), [query]);

  const handleConfirmBulkUnpublish = async () => {
    const data = await bulkUnpublishAction({ documentIds, model, params });
    if (!('error' in data)) {
      selectRow([]);
    }
  };

  const showUnpublishButton =
    hasDraftAndPublishEnabled &&
    hasPublishPermission &&
    documents.some((entry) => entry.status === 'published' || entry.status === 'modified');

  if (!showUnpublishButton) return null;

  return {
    variant: 'tertiary',
    label: formatMessage({ id: 'app.utils.unpublish', defaultMessage: 'Unpublish' }),
    dialog: {
      type: 'dialog',
      title: formatMessage({
        id: 'app.components.ConfirmDialog.title',
        defaultMessage: 'Confirmation',
      }),
      loading: isLoading,
      content: (
        <Flex direction="column" alignItems="stretch" gap={2}>
          <Flex justifyContent="center">
            <WarningCircle width="24px" height="24px" fill="danger600" />
          </Flex>
          <Typography id="confirm-description" textAlign="center">
            {formatMessage({
              id: 'popUpWarning.bodyMessage.contentType.unpublish.all',
              defaultMessage: 'Are you sure you want to unpublish these entries?',
            })}
          </Typography>
          {hasI18nEnabled && (
            <Box textAlign="center" padding={3}>
              <Typography textColor="danger500">
                {formatMessage(
                  {
                    id: getTranslation('Settings.list.actions.unpublishAdditionalInfos'),
                    defaultMessage:
                      'This will unpublish the active locale versions <em>(from Internationalization)</em>',
                  },
                  {
                    em: Emphasis,
                  }
                )}
              </Typography>
            </Box>
          )}
        </Flex>
      ),
      confirmButton: formatMessage({
        id: 'app.utils.unpublish',
        defaultMessage: 'Unpublish',
      }),
      onConfirm: handleConfirmBulkUnpublish,
    },
  };
};

UnpublishAction.type = 'unpublish';

const Emphasis = (chunks: React.ReactNode) => (
  <Typography fontWeight="semiBold" textColor="danger500">
    {chunks}
  </Typography>
);

const DEFAULT_BULK_ACTIONS: BulkActionComponent[] = [PublishAction, UnpublishAction, DeleteAction];

export { DEFAULT_BULK_ACTIONS, BulkActionsRenderer, Emphasis };
export type { BulkActionDescription };
