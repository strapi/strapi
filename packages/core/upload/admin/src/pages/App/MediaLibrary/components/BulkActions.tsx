import { Flex, Typography } from '@strapi/design-system';
import { useIntl } from 'react-intl';

import { getTrad } from '../../../../utils';

import { BulkDeleteButton } from './BulkDeleteButton';
import { BulkMoveButton } from './BulkMoveButton';

import type { File } from '../../../../../../shared/contracts/files';
import type {
  FolderDefinition,
  Folder as FolderInitial,
} from '../../../../../../shared/contracts/folders';

interface FolderWithType extends FolderInitial {
  type: string;
}

export interface FileWithType extends File {
  type: string;
}

export interface BulkActionsProps {
  selected: Array<FileWithType | FolderDefinition> | Array<FolderWithType | FileWithType>;
  onSuccess: () => void;
  currentFolder?: FolderWithType;
}

export const BulkActions = ({ selected = [], onSuccess, currentFolder }: BulkActionsProps) => {
  const { formatMessage } = useIntl();
  const numberAssets = selected?.reduce(function (_this, val) {
    return val?.type === 'folder' && 'files' in val && val?.files && 'count' in val.files
      ? _this + val?.files?.count
      : _this + 1;
  }, 0);

  return (
    <Flex gap={2} paddingBottom={5}>
      <Typography variant="epsilon" textColor="neutral600">
        {formatMessage(
          {
            id: getTrad('list.assets.selected'),
            defaultMessage:
              '{numberFolders, plural, one {1 folder} other {# folders}} - {numberAssets, plural, one {1 asset} other {# assets}} selected',
          },
          {
            numberFolders: selected?.filter(({ type }) => type === 'folder').length,
            numberAssets,
          }
        )}
      </Typography>

      <BulkDeleteButton
        selected={selected as Array<FileWithType | FolderDefinition>}
        onSuccess={onSuccess}
      />
      <BulkMoveButton
        currentFolder={currentFolder}
        selected={selected as Array<FolderWithType | FileWithType>}
        onSuccess={onSuccess}
      />
    </Flex>
  );
};
