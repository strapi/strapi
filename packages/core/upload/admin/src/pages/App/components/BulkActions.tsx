import { Flex, Typography } from '@strapi/design-system';
import { useIntl } from 'react-intl';

import { getTrad } from '../../../utils';

import { BulkDeleteButton } from './BulkDeleteButton';
import { BulkMoveButton } from './BulkMoveButton';

import type { File } from '../../../../../shared/contracts/files';
import type {
  FolderDefinition,
  Folder as FolderInitial,
} from '../../../../../shared/contracts/folders';
interface FolderWithType extends FolderInitial {
  type: string;
}

export interface FileWithType extends File {
  type: string;
}

interface BulkActionsProps {
  selected: Array<FileWithType | FolderDefinition> | Array<FolderWithType | FileWithType>;
  onSuccess: () => void;
  currentFolder?: FolderWithType;
}

export const BulkActions = ({ selected = [], onSuccess, currentFolder }: BulkActionsProps) => {
  const { formatMessage } = useIntl();

  return (
    <Flex gap={2} paddingBottom={5}>
      <Typography variant="epsilon" textColor="neutral600">
        {formatMessage(
          {
            id: getTrad('list.assets.selected'),
            defaultMessage:
              '{numberFolders, plural, one {1 folder} other {# folders}} - {numberAssets, plural, one {1 asset} other {# assets}} selected',
          },
          {
            numberFolders: selected.filter(({ type }) => type === 'folder').length,
            numberAssets: selected.filter(({ type }) => type === 'asset').length,
          }
        )}
      </Typography>

      <BulkDeleteButton
        selected={selected as Array<FileWithType | FolderDefinition>}
        onSuccess={onSuccess}
      />
      <BulkMoveButton
        currentFolder={currentFolder}
        selected={selected as Array<FolderWithType | FileWithType>}
        onSuccess={onSuccess}
      />
    </Flex>
  );
};
