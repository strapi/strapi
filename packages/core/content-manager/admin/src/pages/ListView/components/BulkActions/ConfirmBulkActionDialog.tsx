import * as React from 'react';

import {
  useTable,
  useNotification,
  useAPIErrorHandler,
  useQueryParams,
} from '@strapi/admin/strapi-admin';
import { Button, Flex, Dialog, Typography } from '@strapi/design-system';
import { Check, WarningCircle } from '@strapi/icons';
import { useIntl } from 'react-intl';

import { useDoc } from '../../../../hooks/useDocument';
import { useGetManyDraftRelationCountQuery } from '../../../../services/documents';
import { getTranslation } from '../../../../utils/translations';

import { Emphasis } from './Actions';

interface ConfirmBulkActionDialogProps {
  endAction: React.ReactNode;
  onToggleDialog: () => void;
  isOpen?: boolean;
  dialogBody: React.ReactNode;
}

const ConfirmBulkActionDialog = ({
  onToggleDialog,
  isOpen = false,
  dialogBody,
  endAction,
}: ConfirmBulkActionDialogProps) => {
  const { formatMessage } = useIntl();

  return (
    <Dialog.Root open={isOpen}>
      <Dialog.Content>
        <Dialog.Header>
          {formatMessage({
            id: 'app.components.ConfirmDialog.title',
            defaultMessage: 'Confirmation',
          })}
        </Dialog.Header>
        <Dialog.Body>
          <Flex direction="column" alignItems="stretch" gap={2}>
            <Flex justifyContent="center">
              <WarningCircle width="24px" height="24px" fill="danger600" />
            </Flex>
            {dialogBody}
          </Flex>
        </Dialog.Body>
        <Dialog.Footer>
          <Dialog.Cancel>
            <Button width={'50%'} onClick={onToggleDialog} variant="tertiary">
              {formatMessage({
                id: 'app.components.Button.cancel',
                defaultMessage: 'Cancel',
              })}
            </Button>
          </Dialog.Cancel>
          {endAction}
        </Dialog.Footer>
      </Dialog.Content>
    </Dialog.Root>
  );
};

/* -------------------------------------------------------------------------------------------------
 * BoldChunk
 * -----------------------------------------------------------------------------------------------*/

const BoldChunk = (chunks: React.ReactNode) => <Typography fontWeight="bold">{chunks}</Typography>;

/* -------------------------------------------------------------------------------------------------
 * ConfirmDialogPublishAll
 * -----------------------------------------------------------------------------------------------*/

interface ConfirmDialogPublishAllProps
  extends Pick<ConfirmBulkActionDialogProps, 'isOpen' | 'onToggleDialog'> {
  isConfirmButtonLoading?: boolean;
  onConfirm: () => void;
}

const ConfirmDialogPublishAll = ({
  isOpen,
  onToggleDialog,
  isConfirmButtonLoading = false,
  onConfirm,
}: ConfirmDialogPublishAllProps) => {
  const { formatMessage } = useIntl();
  const selectedEntries = useTable('ConfirmDialogPublishAll', (state) => state.selectedRows);
  const { toggleNotification } = useNotification();
  const { _unstableFormatAPIError: formatAPIError } = useAPIErrorHandler(getTranslation);
  const { model, schema } = useDoc();
  const [{ query }] = useQueryParams<{
    plugins?: {
      i18n?: {
        locale?: string;
      };
    };
  }>();

  // TODO skipping this for now as there is a bug with the draft relation count that will be worked on separately
  // see RFC "Count draft relations" in Notion
  const enableDraftRelationsCount = false;

  const {
    data: countDraftRelations = 0,
    isLoading,
    error,
  } = useGetManyDraftRelationCountQuery(
    {
      model,
      documentIds: selectedEntries.map((entry) => entry.documentId),
      locale: query?.plugins?.i18n?.locale,
    },
    {
      skip: !enableDraftRelationsCount || selectedEntries.length === 0,
    }
  );

  React.useEffect(() => {
    if (error) {
      toggleNotification({ type: 'danger', message: formatAPIError(error) });
    }
  }, [error, formatAPIError, toggleNotification]);

  if (error) {
    return null;
  }

  return (
    <ConfirmBulkActionDialog
      isOpen={isOpen && !isLoading}
      onToggleDialog={onToggleDialog}
      dialogBody={
        <>
          <Typography id="confirm-description" textAlign="center">
            {countDraftRelations > 0 &&
              formatMessage(
                {
                  id: getTranslation(`popUpwarning.warning.bulk-has-draft-relations.message`),
                  defaultMessage:
                    '<b>{count} {count, plural, one { relation } other { relations } } out of {entities} { entities, plural, one { entry } other { entries } } {count, plural, one { is } other { are } }</b> not published yet and might lead to unexpected behavior. ',
                },
                {
                  b: BoldChunk,
                  count: countDraftRelations,
                  entities: selectedEntries.length,
                }
              )}
            {formatMessage({
              id: getTranslation('popUpWarning.bodyMessage.contentType.publish.all'),
              defaultMessage: 'Are you sure you want to publish these entries?',
            })}
          </Typography>
          {schema?.pluginOptions &&
            'i18n' in schema.pluginOptions &&
            schema?.pluginOptions.i18n && (
              <Typography textColor="danger500" textAlign="center">
                {formatMessage(
                  {
                    id: getTranslation('Settings.list.actions.publishAdditionalInfos'),
                    defaultMessage:
                      'This will publish the active locale versions <em>(from Internationalization)</em>',
                  },
                  {
                    em: Emphasis,
                  }
                )}
              </Typography>
            )}
        </>
      }
      endAction={
        <Button
          width={'50%'}
          onClick={onConfirm}
          variant="secondary"
          startIcon={<Check />}
          loading={isConfirmButtonLoading}
        >
          {formatMessage({
            id: 'app.utils.publish',
            defaultMessage: 'Publish',
          })}
        </Button>
      }
    />
  );
};

export { ConfirmDialogPublishAll, ConfirmBulkActionDialog };
export type { ConfirmDialogPublishAllProps, ConfirmBulkActionDialogProps };
