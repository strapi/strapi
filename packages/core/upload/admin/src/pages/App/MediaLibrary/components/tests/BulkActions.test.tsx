import { DesignSystemProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';

import { BulkActions } from '../BulkActions';

import type { BulkActionsProps } from '../BulkActions';

jest.mock('../../../../../hooks/useBulkRemove');
jest.mock('../../../../../components/BulkMoveDialog/BulkMoveDialog');

const setup = (
  props: BulkActionsProps = {
    selected: [],
    onSuccess: jest.fn(),
  }
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        refetchOnWindowFocus: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>
      <DesignSystemProvider>
        <MemoryRouter>
          <IntlProvider locale="en">
            <BulkActions {...props} />
          </IntlProvider>
        </MemoryRouter>
      </DesignSystemProvider>
    </QueryClientProvider>
  );
};

describe('BulkActions', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders', async () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  test('displays the proper assets and folders count', () => {
    const ASSET_COUNT = 2;
    const FOLDER_COUNT = 8;

    const { getByText } = setup({
      onSuccess: jest.fn(),
      selected: [
        ...[...Array(ASSET_COUNT).keys()].map((index) => ({ id: index, type: 'asset' })),
        ...[...Array(FOLDER_COUNT).keys()].map((index) => ({
          id: index,
          type: 'folder',
          files: { count: 0 },
        })),
      ] as BulkActionsProps['selected'],
    });

    expect(
      getByText(`${FOLDER_COUNT} folders - ${ASSET_COUNT} assets selected`)
    ).toBeInTheDocument();
  });
});
