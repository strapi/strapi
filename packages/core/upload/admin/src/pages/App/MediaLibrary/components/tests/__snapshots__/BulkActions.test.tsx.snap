// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkActions renders 1`] = `
.c0 {
  padding-block-end: 20px;
}

.c3 {
  padding-inline-start: 16px;
  padding-inline-end: 16px;
  border-radius: 4px;
  cursor: pointer;
}

.c1 {
  gap: 8px;
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c4 {
  gap: 8px;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: inline-flex;
}

.c6 {
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c2 {
  font-size: 1.6rem;
  line-height: 1.5;
  color: #666687;
}

.c7 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: currentcolor;
  font-weight: 600;
}

.c9 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  height: 3.2rem;
  text-decoration: none;
  border: 1px solid #f5c0b8;
  background: #fcecea;
  color: #b72b1a;
}

.c5:hover {
  background-color: #ffffff;
}

.c5:active {
  background-color: #ffffff;
  border: 1px solid #d02b20;
  color: #d02b20;
}

.c5[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
  cursor: default;
}

.c8 {
  height: 3.2rem;
  text-decoration: none;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
  color: #271fe0;
}

.c8:hover {
  background-color: #ffffff;
}

.c8:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
  color: #4945ff;
}

.c8[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
  cursor: default;
}

@media (prefers-reduced-motion: no-preference) {
  .c5 {
    transition: background-color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),border-color 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
  }
}

@media (prefers-reduced-motion: no-preference) {
  .c8 {
    transition: background-color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),color 120ms cubic-bezier(0.25, 0.46, 0.45, 0.94),border-color 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
  }
}

<div>
  <div
    class="c0 c1"
  >
    <span
      class="c2"
    >
      0 folders - 0 assets selected
    </span>
    <button
      aria-controls="radix-:r0:"
      aria-disabled="false"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="c3 c4 c5"
      data-state="closed"
      type="button"
    >
      <span
        aria-hidden="true"
        class="c6"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 32 32"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 6h-5V5a3 3 0 0 0-3-3h-6a3 3 0 0 0-3 3v1H5a1 1 0 0 0 0 2h1v18a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V8h1a1 1 0 1 0 0-2M14 21a1 1 0 0 1-2 0v-8a1 1 0 0 1 2 0zm6 0a1 1 0 0 1-2 0v-8a1 1 0 0 1 2 0zm0-15h-8V5a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1z"
          />
        </svg>
      </span>
      <span
        class="c7"
      >
        Delete
      </span>
    </button>
    <button
      aria-controls="radix-:r3:"
      aria-disabled="false"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="c3 c4 c8"
      data-state="closed"
      type="button"
    >
      <span
        aria-hidden="true"
        class="c6"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 32 32"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 9H16.414L13 5.586A1.98 1.98 0 0 0 11.586 5H5a2 2 0 0 0-2 2v18.078A1.926 1.926 0 0 0 4.924 27H27.11A1.89 1.89 0 0 0 29 25.111V11a2 2 0 0 0-2-2M5 7h6.586l2 2H5z"
          />
        </svg>
      </span>
      <span
        class="c7"
      >
        Move
      </span>
    </button>
  </div>
  <span
    class="c9"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
